Support config files installation (#472)
eclipse-ankaios/ankaios: doc/docs/reference/config-files.md

# Ankaios Configuration Files

Ankaios now supports configuration files to manage settings more efficiently. These configuration files are optional but provide a convenient way to manage parameters that you may not want to pass as command-line arguments. The priority for parameters is as follows:

1. Command-line arguments
2. Environment variables
3. Configuration file

The configuration files are formatted in TOML, which is easy to parse and fast to process.

## Configuration File Locations

> 

Reviewer: Maybe add some more information like:

```suggestion

The Ankaios configuration files are per default loaded from the following location:

```

- **Server Configuration**: `/etc/ankaios/server.toml`
- **Agent Configuration**: `/etc/ankaios/server.toml`
- **CLI Configuration**: `$HOME/.config/ankaios/ank.toml`

## Configuration File Structure

> 

Reviewer: ```suggestion

The following three examples show how the Ankaios configuration files look like:

```

# The timeout in milliseconds to wait for a response from the ank-server.
response_timeout = 3000  # milliseconds

# The flag to enable verbose output.
verbose = false

# The flag to disable all output.
quiet = false

# The flag that enables waiting for workloads to be created/deleted.
no_wait = false

[default]
# The URL to Ankaios server.
server_url = 'https://127.0.0.1:25551'

# The flag to disable TLS communication with the server.
insecure = false

# The path to the ca certificate pem file.
ca_pem = '/etc/ankaios/certs/ca.pem'

# The path to CLI certificate pem file.
crt_pem = '/home/ankaios/.config/ankaios/ank.pem'

# The path to CLI key pem file.
key_pem = '${XDG_CONFIG_HOME:-$HOME/.config}/ankaios/ank-key.pem'

# The content of the ca certificate pem file (optional).
# ca_pem_content = '''-----BEGIN CERTIFICATE-----
# ...
# -----END CERTIFICATE-----'''

# The content of the CLI certificate pem file (optional).
# crt_pem_content = '''-----BEGIN CERTIFICATE-----
# ...
# -----END CERTIFICATE-----'''

# The content of the CLI key pem file (optional).
# key_pem_content = '''-----BEGIN PRIVATE KEY-----
# ...
# -----END PRIVATE KEY-----'''
```

## Using the Configuration Files

> To use the configuration files, you can specify the path to the configuration file using the `-x` command-line argument:

Reviewer: ```suggestion
To use the configuration files, just place a config file at the default location specified [above](#configuration-file-locations), or specify the path to the configuration file using the `-x` command-line argument:
```

```sh
ank-server -x /path/to/ank-server.conf
ank-agent -x /path/to/ank-agent.conf
ank -x /path/to/ank.conf
```

## Notes

- The configuration files are optional. If not provided, the default values will be used.
- You can specify either the path to the certificate files or the content of the certificate files, but not both.
- The `version` field is mandatory to ensure compatibility.

> This documentation provides an introduction in the usage of config files for Ankaios components. Please fell free to contact us if you have any
> any questions or need further assistance.

Reviewer: ```suggestion
This documentation provides an introduction in the usage of config files for Ankaios components. Please fell free to contact us if you have
any questions or need further assistance.
```
